src/gwt/src/org/rstudio/studio/client/RStudio.java

```diff
@@ -155,8 +155,47 @@ public void execute()
       };
    }
    
+   private static final native void maybeDelayLoadApplication(RStudio rstudio)
+   /*-{
+   	
+      if ($wnd.qt)
+      {
+         // on the desktop main window, we may need to wait for Qt to finish
+         // initialization before loading GWT
+         if ($wnd.qt.webChannelReady)
+         {
+            // Qt is ready; load the application now
+            rstudio.@org.rstudio.studio.client.RStudio::delayLoadApplication()();
+            return;
+         }
+         
+         // Qt not yet ready; set a hook and let the Qt WebChannel
+         // initialization script call it to finish initialization
+         $wnd.delayLoadApplication = $entry(function() {
+            rstudio.@org.rstudio.studio.client.RStudio::delayLoadApplication()();
+         });
+         
+         // set a timeout and attempt load just in case something goes wrong with
+         // Qt initialization (we don't want to just leave the user with a blank
+         // window)
+         setTimeout(function() {
+            if ($wnd.delayLoadApplication) {
+               $wnd.delayLoadApplication();
+               $wnd.delayLoadApplication = null;
```

Comment:
maybe set this to `null` inside `$wnd.delayLoadApplication` so we don't have to remember to clear it in both places?